cucumber-expressions/ruby/lib/cucumber/cucumber_expressions/parameter_type_matcher.rb

```diff
@@ -3,26 +3,30 @@ module CucumberExpressions
     class ParameterTypeMatcher
       attr_reader :parameter_type
 
-      def initialize(parameter_type, regexp, text, match_position=0)
-        @parameter_type, @regexp, @text = parameter_type, regexp, text
+      def initialize(parameter_type, regexp, text, match_position=0, force_full_word=true)
+        @parameter_type, @regexp, @text, @force_full_word = parameter_type, regexp, text, force_full_word
         @match = @regexp.match(@text, match_position)
       end
 
       def advance_to(new_match_position)
         (new_match_position...@text.length).each {|advancedPos|
-          matcher = self.class.new(parameter_type, @regexp, @text, advancedPos)
-          if matcher.find
+          matcher = self.class.new(parameter_type, @regexp, @text, advancedPos, @force_full_word)
+          if matcher.find && matcher.full_word?
```

Comment:
I would have expected this flag to be used in the Cucumber Expression Generator. What is the effect of using it here?

Reply:
I get the impression this actually changes how the matching works.

Reply:
Owh wait. I get it now. It's matcher. Not the parameter type.